pkg/server/vault.go

```diff
@@ -0,0 +1,405 @@
+		return nil, &VaultError{Message: fmt.Sprintf(VaultErrorReadResult, vErr)}
+	}
+
+	data, ok := secret.Data["data"].(map[string]interface{})
+	if !ok {
+		return nil, &VaultError{Message: fmt.Sprintf(VaultErrorReadResult, vErr)}
+	}
+	values := make([]interface{}, 0, len(data))
+	for _, value := range data {
+		values = append(values, value)
+	}
+	stringValues := []string{}
+	for _, value := range data {
+		if str, ok := value.(string); ok {
+			stringValues = append(stringValues, str)
+		}
+	}
+	return stringValues, nil
+}
+
+// DeleteSecret removes a specific key from a stored secret in the Vault secrets engine.
+//
+// This function reads the existing secret data from Vault, removes the specified key,
+// and updates the stored secret. It is designed for use with a local Vault running
+// in development mode.
+//
+// Parameters:
+// - path: The Vault path where the secret is stored (e.g., "secret/myapp").
+// - key: The specific key within the secret that should be removed.
+//
+// Returns:
+// - *VaultError: An error object if the operation fails; otherwise, nil.
+func (v *VaultClient) DeleteSecret(path, key string) *VaultError {
+	if err := func() *VaultError {
+		switch {
+		case v.client == nil:
+			return &VaultError{Message: fmt.Sprintf(VaultErrorClient)}
+		case len(path) == 0:
+			return &VaultError{Message: fmt.Sprintf(VaultErrorNoKeyPath)}
+		case len(key) == 0:
+			return &VaultError{Message: fmt.Sprintf(VaultErrorNoKeyName)}
+		default:
+			return nil
+		}
+	}(); err != nil {
+		return err
+	}
+	client := v.client
+
+	// Read existing data
+	secret, vErr := client.Logical().Read(path)
+	if vErr != nil || secret == nil {
+		return &VaultError{Message: fmt.Sprintf(VaultErrorReadResult, vErr)}
+	}
+
+	// Remove key from data
+	data := secret.Data["data"].(map[string]interface{})
+	if _, exists := data[key]; exists {
+		delete(data, key)
+	} else {
+		return &VaultError{Message: fmt.Sprintf(VaultErrorResultNotExist, key)}
+	}
+
+	// Write updated data back to Vault
+	updatedSecret := map[string]interface{}{
+		"data": data,
+	}
+
+	_, vErr = client.Logical().Write("secret/data/"+path, updatedSecret)
+	if vErr != nil {
+		return &VaultError{Message: fmt.Sprintf(VaultErrorUpdate, key)}
+	}
+	return nil
+}
+
+// CloseClient stops the running Vault process on the local machine in development mode.
+//
+// This function detects the operating system and executes the appropriate command
+// to terminate the Vault process. It is intended for use in development mode when
+// Vault is running locally.
+//
+// - On Windows, it uses PowerShell to forcefully stop the Vault process.
+// - On Linux/macOS, it uses the `pkill` command to terminate Vault.
+//
+// This function does not return an error but logs the status of the termination.
+func (v *VaultClient) CloseClient() {
+	if runtime.GOOS == "windows" {
+		// Run PowerShell command to force kill Vault
+		cmd := exec.Command("powershell", "-Command", "Get-Process vault | Stop-Process -Force")
```

Comment:
We have implemented `make setup` and `make teardown` elsewhere. Let's use that outside of Go code.

https://github.com/moov-io/base/blob/master/makefile#L2
https://github.com/moov-io/base/blob/master/.github/workflows/nocgo.yml#L33

Github's macos and windows runners don't always support docker, so check `testing.Short()` in tests that need an external resource. 

Do not copy CGO variables in Github actions from the moov-io/base workflows.

Reply:
I removed code in new commit